File: 2-Ad-Hoc/Ex1130.java
import java.io.InputStreamReader;
import java.io.IOException;
import java.io.BufferedReader;

public class Ex1130 {
    static boolean jogada(StringBuffer tabuleiro) {
        if (tabuleiro.indexOf("XXX") != -1) {
            return true;
        } else if (tabuleiro.indexOf("...") == -1 && tabuleiro.indexOf("X.X") == -1 && tabuleiro.indexOf(".XX") == -1
                && tabuleiro.indexOf("XX.") == -1) {
            return true;
        }

        return false;
    }

    public static void main(String args[]) throws IOException {
        /* variaveis */
        InputStreamReader in = new InputStreamReader(System.in);
        BufferedReader entrada = new BufferedReader(in);
        String aux = new String();
        int n;
        char ePossivel;

        /* entrada */
        while (true) {
            n = entrada.read();
            if (n == '0') { // condicao de parada
                break;
            }

            ePossivel = 'N';
            StringBuffer tabuleiro = new StringBuffer(n), tabFuturo = new StringBuffer(n);

            entrada.readLine();
            aux = entrada.readLine();
            tabuleiro.append(aux);

            // processamento
            for (int i = 0; i < tabuleiro.length(); i++) {
                tabFuturo.delete(0, tabFuturo.length());
                tabFuturo.append(aux);
                if (tabuleiro.charAt(i) == '.') {
                    tabFuturo.setCharAt(i, 'X');
                    if (jogada(tabFuturo)) {
                        ePossivel = 'S';
                        break;
                    }
                }
            }
            // saida
            System.out.println(ePossivel);
        }

        entrada.close();
    }
}

File: 2-Ad-Hoc/ex1130.py
def situacoesProibidas(tab):
    situacoes = ('...', 'X.X', '.XX', 'XX.')
    for i in range(0, 4):
        if situacoes[i] in tab:
            return False
    return True
    
#entrada
while True:
    n = int(input())
    if n == 0: #condicao de parada
        break
    tabuleiro = str(input())

    #processamento
    ePossivel = 'N'

    for i in range(0, n):
        tabFuturo = ''
        if tabuleiro[i] == '.':
            tabFuturo = tabFuturo + tabuleiro[0:i] + 'X' + tabuleiro[i + 1:]
            if ('XXX' in tabFuturo) or situacoesProibidas(tabFuturo):
                ePossivel = 'S'
                break

    #saida
    print(ePossivel)
